### DISCUSSION TITLE
父级菜单不见了

### DISCUSSION THREAD

**Discussion by cj814** [umijs/umi#11499] [Q&A] [answered]
登录以后，左侧父级菜单默认不显示，需要点一下右侧SettingDrawer组件的“侧边菜单布局”才出现父级菜单，config/defaultSettings.ts的layout属性已经设置了side。问题复现demo：https://github.com/cj814/umi-test

![cf79a50e_1760085](https://github.com/umijs/umi/assets/28130645/0ef8ce3d-e5c8-486f-abe5-7bfa9995e3d2)
![46ec7005_1760085](https://github.com/umijs/umi/assets/28130645/d82345f6-8fe9-4074-819f-7dd09400df3d)
![85b2ab10_1760085](https://github.com/umijs/umi/assets/28130645/e7d4d594-a73f-4f2c-988f-2817d31cafc3)
![064627d0_1760085](https://github.com/umijs/umi/assets/28130645/cb549613-3550-4bf7-b7fb-566dcb16b240)

**Comment by fz6m** [comment]
动态添加路由表的时候有问题。

不支持动态导入拼接 `import('@/pages/' + path)` ，webpack 不知道要打包哪些文件，需要维护一个所有路由表的映射，然后根据 key 来动态添加路由表。

可以参考 https://github.com/umijs/umi/discussions/9717#discussioncomment-4078053 动态添加路由。

你可以从 `@umijs/max` 导入 `useAppData` 这个 hook 打印看看你的路由有没有生效，在你的这个 repo 里是没有生效的，自己排查下吧。

如还有问题，需要给一个有问题的最小复现。

**Reply by cj814** [reply, marked as answer]
我找到问题了，这个动态拼接是支持的，问题出在默认设置了splitMenus: true，改成false就好了
